nflow-engine/src/main/java/io/nflow/engine/internal/executor/WorkflowStateProcessor.java

```diff
@@ -195,6 +200,32 @@ private void runImpl() {
     logger.debug("Finished.");
   }
 
+  private void logRetryableException(WorkflowState state, Throwable t) {
+    ExceptionSeverity exceptionSeverity = state.getExceptionSeverity(t);
+    BiConsumer<String, Object> logMethod = getLogMethod(exceptionSeverity.logLevel);
+    if (exceptionSeverity.logStackTrace) {
+      logMethod.accept("Handler threw a retryable exception, trying again later.", t);
+    } else {
+      logMethod.accept("Handler threw a retryable exception, trying again later. Message: {}", t.getMessage());
+    }
+  }
+
+  private BiConsumer<String, Object> getLogMethod(Level logLevel) {
+    switch (logLevel) {
```

Comment:
Does the logger not have a generic logger method that takes the level as first argument?

Reply:
According to this, no: https://stackoverflow.com/questions/2621701/setting-log-level-of-message-at-runtime-in-slf4j